packages/davinci-client/src/lib/client.store.ts

```diff
@@ -46,26 +48,40 @@ import { NodeStates, StartNode } from '../types.js';
 export async function davinci<ActionType extends ActionTypes = ActionTypes>({
   config,
   requestMiddleware,
+  logger,
 }: {
   config: DaVinciConfig;
   requestMiddleware?: RequestMiddleware<ActionType>[];
+  logger?: {
+    level: LogLevel;
+    custom: ReturnType<typeof loggerFn>;
+  };
 }) {
-  const store = createClientStore({ requestMiddleware });
+  const log = loggerFn({ level: logger?.level || 'error', custom: logger?.custom });
+  const store = createClientStore({ requestMiddleware, logger: log });
 
   if (!config.serverConfig.wellknown) {
-    throw new Error('`wellknown` property is a required as part of the `config.serverOptions`');
+    const error = new Error(
+      '`wellknown` property is a required as part of the `config.serverConfig`',
+    );
+    log.error(error.message);
+    throw error;
   }
 
   if (!config.clientId) {
-    throw new Error('`clientId` property is a required as part of the `config`');
+    const error = new Error('`clientId` property is a required as part of the `config`');
```

Comment:
If this isn't what your thinking feel free to not implement but what if we had a function that did this?

```ts
const logAndThrow = message => pipe(
  message,
  () => new Error(message)
  (error) => {
     log.error(error.message);
     return error
     }, // could probably be better replaced by `tap` if we write it out.
     () => throw error
    );
```

we can write our own `pipe` if something like redux doesn't have it available. I think base redux does, but not RTK

Reply:
Okay, let me look into this.

Reply:
I didn't use `pipe`, but I created a higher order function to abstract out some of the logic. Didn't address this stuff that threw Errors, but I did addressed the redundant stuff in `update` and `validate`.